Make sure RPATH is enabled regardless of the OS detection.
--- src/src/rpath.pri	2020-07-07 06:44:20.000000000 +0000
+++ src.mod/src/rpath.pri	2021-05-24 14:00:48.228091000 +0000
@@ -8,7 +8,7 @@
 
 macos {
     QMAKE_LFLAGS += -Wl,-rpath,@loader_path/$$REL_PATH_TO_LIBS,-rpath,@loader_path/$$REL_PATH_TO_PLUGINS
-} else:linux-* {
+} else {
     QMAKE_RPATHDIR += \$\$ORIGIN
     QMAKE_RPATHDIR += \$\$ORIGIN/$$REL_PATH_TO_LIBS
     QMAKE_RPATHDIR += \$\$ORIGIN/$$REL_PATH_TO_PLUGINS
--- src/src/qtcreatorplugin.pri	2020-07-07 06:44:20.000000000 +0000
+++ src.mod/src/qtcreatorplugin.pri	2021-05-24 14:01:30.264902000 +0000
@@ -96,7 +96,7 @@
 
 TEMPLATE = lib
 CONFIG += plugin plugin_with_soname
-linux*:QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
+QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
 
 target.path = $$INSTALL_PLUGIN_PATH
 INSTALLS += target
--- src/src/tools/qtcreatorwidgets/qtcreatorwidgets.pro	2020-07-07 06:44:20.000000000 +0000
+++ src.mod/src/tools/qtcreatorwidgets/qtcreatorwidgets.pro	2021-05-24 14:02:28.788142000 +0000
@@ -13,17 +13,13 @@
     IDE_LIBRARY_BASENAME = lib
 }
 
-linux-*||win32 {
-  # form abs path to qtcreator lib dir
-  QTC_LIBS=$$dirname(OUT_PWD)
-  QTC_LIBS=$$dirname(QTC_LIBS)
-  QTC_LIBS=$$dirname(QTC_LIBS)
-  QTC_LIBS=$$QTC_LIBS/$$IDE_LIBRARY_BASENAME/qtcreator
-}
+# form abs path to qtcreator lib dir
+QTC_LIBS=$$dirname(OUT_PWD)
+QTC_LIBS=$$dirname(QTC_LIBS)
+QTC_LIBS=$$dirname(QTC_LIBS)
+QTC_LIBS=$$QTC_LIBS/$$IDE_LIBRARY_BASENAME/qtcreator
 
-linux-*{  
-  QMAKE_RPATHDIR *= $$QTC_LIBS
-}
+QMAKE_RPATHDIR *= $$QTC_LIBS
 
 INCLUDEPATH += ../../../src/libs
 macx {
